conda/resolve.py

```diff
@@ -896,6 +920,26 @@ def mysat(specs, add_if=False):
             solution, obj7 = C.minimize(eq_optional_c, solution)
             log.debug('Package removal metric: %d', obj7)
 
+            # Requested packages (split by features): maximize versions
+            eq_req_c_f, eq_req_v_f, eq_req_b_f, eq_req_t_f = (
+                r2.generate_version_metrics(C, specr, split_by_features=True))
+            solution, obj3a_f = C.minimize(eq_req_c_f, solution)
+            solution, obj3_f = C.minimize(eq_req_v_f, solution)
+            log.debug('Initial package channel/version metric (feature split): %d/%d',
+                      obj3a_f, obj3_f)
+            # Requested packages (split by features): maximize builds
+            solution, obj4_f = C.minimize(eq_req_b_f, solution)
+            log.debug('Initial package build metric (feature split): %d', obj4_f)
+
+            # Remaining packages (split by features): maximize versions, then builds
+            eq_c_f, eq_v_f, eq_b_f, eq_t_f = (
+                r2.generate_version_metrics(C, speca, split_by_features=True))
+            solution, obj5a_f = C.minimize(eq_c_f, solution)
+            solution, obj5_f = C.minimize(eq_v_f, solution)
+            solution, obj6_f = C.minimize(eq_b_f, solution)
+            log.debug('Additional package channel/version/build metrics (feature split): %d/%d/%d',
+                      obj5a_f, obj5_f, obj6_f)
+
```

Comment:
Interesting, I think I like the idea. A question: since we maximize the version above, can the maximization by version below be removed? Otherwise, it would be duplicate work, wouldn't it?

Reply:
I suppose so. I was currently just too tired to think about the implications, thus I just left those in place.

Reply:
But overall, all the (increasing) "real world examples" I stumbled upon that are blocked by this issue seem to work with those changes.

Reply:
So, the test failure looks like it is now picking up more recent build numbers. Is this related to your modifications or did that happen of 4.3.x branch before?

Reply:
The changes imply that newer builds are more prioritized, yes. Previously, the channel priorities, version and build numbers of the dependencies where only considered much later on. Furthermore, due to the early feature count minimization, more recent builds of requested packages didn't survive before if there were other builds that didn't require additional features.